### ISSUE TITLE
Xenial release nightlies time/clock failures

### ISSUE THREAD

**Issue by dhood** [ros2/build_farmer#147]
These tests started failing august 29:

```
projectroot.cppcheck
projectroot.rclpytests
rclpy.test.test_clock.TestClock.test_triggered_clock_change_callbacks
rclpy.test.test_clock.TestClock.test_triggered_time_jump_callbacks
rclpy.test.test_time_source.TestTimeSource.test_backwards_jump
rclpy.test.test_time_source.TestTimeSource.test_clock_change
rclpy.test.test_time_source.TestTimeSource.test_forwards_jump
rclpy.test.test_time_source.TestTimeSource.test_no_post_callback
rclpy.test.test_time_source.TestTimeSource.test_no_pre_callback
rclcpp.cppcheck.error: deallocret (src/rclcpp/clock.cpp:138)
```

https://ci.ros2.org/view/nightly/job/nightly_xenial_linux_release/82/
https://ci.ros2.org/view/nightly/job/nightly_xenial_linux-aarch64_release/87/

Which was after these PRs were merged (cc @sloretz):
https://github.com/ros2/rclpy/pull/222 (adding some of the tests)
https://github.com/ros2/rclcpp/pull/543
https://github.com/ros2/rcl/pull/284

**Comment by sloretz** [comment]
All tests are fixed by ros2/rclpy#237 and ros2/rclcpp#556. I think this can be closed when both are merged.

**Comment by dhood** [comment]
am going to close this under the assumption the xenial nightlies will go back to green tonight; can reopen it if not.
